commit 5aeafc2b3675b8a7712a1151e6baf457e792d7de (refs/changes/82/5082/1)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-11-21T00:35:52-08:00 (11 months ago)
    
    TRITON-976 agents-installer should have option to just unpack shar without executing install.sh

diff --git a/mk-agents-shar b/mk-agents-shar
index 7158859..c8eeb2f 100755
--- a/mk-agents-shar
+++ b/mk-agents-shar
@@ -219,8 +219,13 @@ function mk_agents_shar() {
 
     (cat <<__EOF__
 #!/bin/sh
-if [ \`pwd\` != '$tmp' ]; then
-  cd $tmp
+if [[ \`pwd\` != '$tmp' ]]; then
+    cd $tmp
+fi
+
+if [[ -n \$UNPACK_SUBDIR ]]; then
+    mkdir -p \$UNPACK_SUBDIR
+    cd \$UNPACK_SUBDIR
 fi
 
 # Clean up possible previous runs
@@ -233,6 +238,10 @@ __EOF__
 set -o errexit
 set -o pipefail
 
+if [[ \$UNPACK_ONLY == "true" ]]; then
+    exit 0
+fi
+
 if [[ -f agents/install.sh ]]; then
     (cd agents && /bin/bash ./install.sh)
 fi
diff --git a/package.json b/package.json
index 7000262..9c289b3 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "agents-installer",
   "description": "Build agentsshar package for SDC",
-  "version": "2.1.0",
+  "version": "2.2.0",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
